Tokio-util should have its tokio dependency bumped

Got the following build error:

```
error[E0433]: failed to resolve: could not find `coop` in `task`
   --> /.../tokio-util-0.7.17/src/task/join_queue.rs:197:46
    |
197 |         let jh = std::pin::pin!(tokio::task::coop::unconstrained(jh));
    |  
```

It looks to be due to tokio-util because the tokio-util tokio dependency is on version 1.28.0 (https://docs.rs/crate/tokio-util/0.7.17/source/Cargo.toml#32), which doesn't have `task::coop` (published in 1.44 — It exists in https://docs.rs/tokio/1.44.0/tokio/task/coop/index.html, but not in https://docs.rs/tokio/1.43.3/tokio/task/coop/index.html). 

This applies only to tokio-util 0.7.17, which introduced `join_queue` (there in https://docs.rs/crate/tokio-util/0.7.17/source/src/task/, not in https://docs.rs/crate/tokio-util/0.7.16/source/src/task/).

I'll put up an issue instead of a fix because I'm not sure if the desired solution is to bump the tokio dependency or to gain `Unconstrained` from `task` for pre 1.44 versions, or to gate `join_queue`.

--- a/tokio-util/Cargo.toml
+++ b/tokio-util/Cargo.toml
@@ -35,7 +35,7 @@ join-map = ["rt", "hashbrown"]
 __docs_rs = ["futures-util"]
 
 [dependencies]
-tokio = { version = "1.28.0", path = "../tokio", features = ["sync"] }
+tokio = { version = "1.44.0", path = "../tokio", features = ["sync"] }
 bytes = "1.5.0"
 futures-core = "0.3.0"
 futures-sink = "0.3.0"

